examples/chip-tool/commands/common/Command.h

```diff
@@ -91,10 +93,13 @@ class Command
     const char * GetName(void) const { return mName; }
     const char * GetAttribute(void) const;
     const char * GetArgumentName(size_t index) const;
+    const char * GetOptionalArgumentName(size_t index) const;
     size_t GetArgumentsCount(void) const { return mArgs.size(); }
+    size_t GetOptionalArgumentsCount(void) const { return mOptionalArgs.size(); }
 
     bool InitArguments(int argc, char ** argv);
-    size_t AddArgument(const char * name, const char * value);
+    bool InitOptionalArguments(int & argc, char ** argv);
```

Comment:
Need to document the semantics, especially of the return value and the in/out parameter.